### ISSUE TITLE
Error: received Ctrl-C or deployments could not stabilize within 10m0s: kubectl rollout status command interrupted.

### ISSUE THREAD

**Issue by Emixam23** [GoogleCloudPlatform/microservices-demo#311]
Hey, 

I have an issue regarding the deployment part (localhost):

```
Starting deploy...
 - deployment.apps/adservice configured
 - service/adservice configured
 - deployment.apps/cartservice configured
 - service/cartservice configured
 - deployment.apps/checkoutservice configured
 - service/checkoutservice configured
 - deployment.apps/currencyservice configured
 - service/currencyservice configured
 - deployment.apps/emailservice configured
 - service/emailservice configured
 - deployment.apps/frontend configured
 - service/frontend configured
 - service/frontend-external configured
 - deployment.apps/loadgenerator configured
 - deployment.apps/paymentservice configured
 - service/paymentservice configured
 - deployment.apps/productcatalogservice configured
 - service/productcatalogservice configured
 - deployment.apps/recommendationservice configured
 - service/recommendationservice configured
 - deployment.apps/redis-cart configured
 - service/redis-cart configured
 - deployment.apps/shippingservice configured
 - service/shippingservice configured
Waiting for deployments to stabilize
 - default:deployment/adservice 
 - default:deployment/cartservice 
 - default:deployment/checkoutservice 
 - default:deployment/currencyservice 
 - default:deployment/emailservice 
 - default:deployment/frontend 
 - default:deployment/loadgenerator 
 - default:deployment/paymentservice 
 - default:deployment/productcatalogservice 
 - default:deployment/recommendationservice 
 - default:deployment/redis-cart 
 - default:deployment/shippingservice 
 - default:deployment/adservice Waiting for deployment "adservice" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/cartservice Waiting for deployment "cartservice" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/checkoutservice Waiting for deployment "checkoutservice" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/currencyservice Waiting for deployment "currencyservice" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/emailservice Waiting for deployment "emailservice" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/frontend Waiting for deployment "frontend" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/loadgenerator Waiting for deployment "loadgenerator" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/paymentservice Waiting for deployment "paymentservice" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/productcatalogservice Waiting for deployment "productcatalogservice" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/recommendationservice Waiting for deployment spec update to be observed...
 - default:deployment/redis-cart Waiting for deployment "redis-cart" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/shippingservice Waiting for deployment spec update to be observed...
 - default:deployment/recommendationservice Waiting for deployment "recommendationservice" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/shippingservice Waiting for deployment "shippingservice" rollout to finish: 1 old replicas are pending termination...
 - default:deployment/checkoutservice is ready. [11/12 deployment(s) still pending]
 - default:deployment/cartservice is ready. [10/12 deployment(s) still pending]
 - default:deployment/currencyservice is ready. [9/12 deployment(s) still pending]
 - default:deployment/paymentservice is ready. [8/12 deployment(s) still pending]
 - default:deployment/productcatalogservice is ready. [7/12 deployment(s) still pending]
 - default:deployment/shippingservice is ready. [6/12 deployment(s) still pending]
 - default:deployment/frontend is ready. [5/12 deployment(s) still pending]
 - default:deployment/adservice failed. [4/12 deployment(s) still pending] Error: Running [kubectl --context docker-desktop rollout status deployment adservice --namespace default --watch=false]
 - stdout: 
 - stderr: error: deployment "adservice" exceeded its progress deadline
: exit status 1.
 - default:deployment/emailservice failed. [3/12 deployment(s) still pending] Error: received Ctrl-C or deployments could not stabilize within 10m0s: kubectl rollout status command interrupted.
 - default:deployment/recommendationservice failed. [2/12 deployment(s) still pending] Error: received Ctrl-C or deployments could not stabilize within 10m0s: kubectl rollout status command interrupted.
 - default:deployment/redis-cart failed. [1/12 deployment(s) still pending] Error: received Ctrl-C or deployments could not stabilize within 10m0s: kubectl rollout status command interrupted.
 - default:deployment/loadgenerator failed. Error: received Ctrl-C or deployments could not stabilize within 10m0s: kubectl rollout status command interrupted.
FATA[0649] 5/12 deployment(s) failed  
```

Can someone explain to me where I am wrong please?

**Comment by ahmetb** [comment]
Not sure, we don't support `kubectl rollout` and the code is open source and short. I recommend you investigate.

**Comment by Emixam23** [comment]
Hey!

Actually I am pretty new, I only download the repository and followed the readme:

```
MacBook-Pro-de-Emixam23:microservices-demo emixam23$ skaffold run --default-repo=gcr.io/<PROJECT_ID>
```

This error came up after 15mins i would say. From your README:
https://github.com/GoogleCloudPlatform/microservices-demo#option-1-running-locally

I am ok to investigate, this seems amazing and might answer a lot of questions I have, but I am not sure to know what I have to do, or in which direction I should go, sorry :/

Thanks!

**Comment by ahmetb** [comment]
Sorry as you can imagine we don't have bandwidth to support everyone's questions.

**Comment by Emixam23** [comment]
Ok I finally found out, actually the issue is that, in local, it tries to reach the profiler and some others resources, however, it can't find the credentials and fails (crashLoopBackOff) and I think, that for some reason, it affects the deployement from `skaffold`.

I so then uncommented all the env variable :
- DISABLE_TRACING
- DISABLE_PROFILER
- etc etc

Now it works like a charm in local. 

I also tried to deploy it without touching nothing to GKE and it worked directly, so I think that my issue was only related to env variables settings. 

Last point, thans to you @ahmetb for pointing out this repository from https://github.com/grpc-ecosystem/grpc-health-probe/issues/40 .. What a gold mine is this repository, thank you so much and thank to all the person who worked on it, super impressive and WAY MORE complet than the officials docs to me :)
